hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/RequestHandler.java

```diff
@@ -121,8 +153,8 @@ public void stop() {
   private boolean isLocalViewBehind(Context ctx) {
     String basePath = ctx.queryParam(RemoteHoodieTableFileSystemView.BASEPATH_PARAM);
     String lastKnownInstantFromClient =
-        ctx.queryParam(RemoteHoodieTableFileSystemView.LAST_INSTANT_TS, HoodieTimeline.INVALID_INSTANT_TS);
-    String timelineHashFromClient = ctx.queryParam(RemoteHoodieTableFileSystemView.TIMELINE_HASH, "");
+        ctx.queryParamAsClass(RemoteHoodieTableFileSystemView.LAST_INSTANT_TS, String.class).getOrDefault(HoodieTimeline.INVALID_INSTANT_TS);
```

Comment:
We should consider wrapping this logic into a util method so that it does not look verbose in the core request handler.  Let's push that refactoring work into a separate PR.